camayoc/types/settings.py

```diff
@@ -144,12 +144,18 @@ class ExpectedOpenShiftData(BaseModel):
     operators: list[str]
 
 
+class ExpectedAggregateData(BaseModel):
+    results: dict[str, Any]
+    diagnostics: dict[str, int]
+
+
 class ExpectedScanData(BaseModel):
     distribution: Optional[ExpectedDistributionData] = None
     products: Optional[list[ExpectedProductData]] = None
     cluster_info: Optional[ExpectedOpenShiftData] = None
     installed_products: Optional[list[str]] = None
     raw_facts: Optional[dict[str, Any]] = None
+    aggregate: Optional[ExpectedAggregateData] = None
```

Comment:
I think that one deserves a comment explaining that `aggregate` contains data about entire scan, not a specific host. This is different from all the other fields here, which contain some data about a specific host. We put `aggregate` at the same level because entire framework is written this way and we have a number of helpers that make it easy to work with data that follows such shape, but technically `aggregate` would fit more at `ScanOptions` level (or entire `ExpectedScanData` should be inverted, so you have report type first, and hostnames inside).

This is more visible in config PR - `aggregate` key is put under `discovery-fuzzy` host, but it contains data about all the hosts within that scan.

Of course you _could_ move `aggregate` to `ScanOptions`, and that will make `test_aggregate_report` API test a bit easier (as you won't have to loop over hosts and skip most of them), but then you won't be able to use `scans.ok_with_expected_data_attr`. So I suggest comment as an option that requires less work.

Reply:
Yes, I do believe the explanation is necessary in the code. Thanks for your feedback.